include/aspect/heating_model/interface.h

```diff
@@ -126,67 +164,132 @@ namespace aspect
-     * from input files.
-     * @param factory_function A pointer to a function that can create an
-     * object of this heating model.
+     * A class that manages all objects that provide functionality to the
+     * heating models.
      *
-     * @ingroup HeatingModels
+     * @ingroup MeshRefinement
      */
     template <int dim>
-    void
-    register_heating_model (const std::string &name,
-                            const std::string &description,
-                            void (*declare_parameters_function) (ParameterHandler &),
-                            Interface<dim> *(*factory_function) ());
+    class Manager : public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Destructor. Made virtual since this class has virtual member
+         * functions.
+         */
+        virtual ~Manager ();
 
-    /**
-     * A function that given the name of a model returns a pointer to an
-     * object that describes it. Ownership of the pointer is transferred to
-     * the caller.
-     *
-     * The model object returned is not yet initialized and has not read its
-     * runtime parameters yet.
-     *
-     * @ingroup HeatingModels
-     */
-    template <int dim>
-    Interface<dim> *
-    create_heating_model (ParameterHandler &prm);
+        /**
+         * Declare the parameters of all known mesh heating plugins, as
+         * well as of ones this class has itself.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
 
-    /**
-     * Return a list of names of all implemented heating models, separated by
-     * '|' so that it can be used in an object of type Patterns::Selection.
-     */
-    template <int dim>
-    std::string
-    get_names ();
 
-    /**
-     * Declare the runtime parameters of the registered heating models.
-     *
-     * @ingroup HeatingModels
-     */
-    template <int dim>
-    void
-    declare_parameters (ParameterHandler &prm);
+        /**
+         * Read the parameters this class declares from the parameter file.
+         * This determines which heating model objects will be created; then
+         * let these objects read their parameters as well.
+         */
+        void
+        parse_parameters (ParameterHandler &prm);
+
+        /**
+         * A function that is called at the beginning of each time step,
+         * calling the update function of the individual heating models.
+         */
+        void
+        update ();
+
+
+        /**
+         * A function that calls the evaluate function of all the individual
+         * heating models.
+         */
+        void
+        execute (const typename aspect::MaterialModel::Interface<dim>::MaterialModelInputs &material_model_inputs,
+                 const typename aspect::MaterialModel::Interface<dim>::MaterialModelOutputs &material_model_outputs,
+                 HeatingModel::HeatingModelOutputs &heating_model_outputs) const;
```

Comment:
We called it execute() to match the name in the mesh refinement plugin, which also combines several plugins, whereas in the material model we always have only one plugin.